src/azure-cli/azure/cli/command_modules/apim/custom.py

```diff
@@ -207,3 +221,87 @@ def update_apim_api(instance, description=None, subscription_key_parameter_names
         instance.tags = tags
 
     return instance
+
+
+def import_apim_api(client, resource_group_name, service_name, api_path, description=None, subscription_key_parameter_names=None, api_revision=None,
+                    api_id=None, api_version=None, api_version_set_id=None, display_name=None, service_url=None, protocols=None, specificationPath=None,
+                    specificationUrl=None, specificationFormat=None, api_type=None, subscription_required=None,
+                    soap_api_type=None, wsdl_endpoint_name=None, wsdl_service_name=None, no_wait=False):
+    """Import a new API"""
+    cms = client.api
+
+    # api_type: Type of API. Possible values include: 'http', 'soap'
+    # possible parameter format is 'wadl-xml', 'wadl-link-json', 'swagger-json', 'swagger-link-json', 'wsdl', 'wsdl-link', 'openapi', 'openapi+json', 'openapi-link'
+    # possible parameter specificationFormat is 'Wadl', 'Swagger', 'OpenApi', 'OpenApiJson', 'Wsdl'
+
+    if specificationFormat == ImportFormat.Wadl.value:
+        sFormat = ContentFormat.wadl_xml.value if specificationPath is not None else ContentFormat.wadl_link_json.value
+    elif specificationFormat == ImportFormat.Swagger.value:
+        sFormat = ContentFormat.swagger_json.value if specificationPath is not None else ContentFormat.swagger_link_json.value
+    elif specificationFormat == ImportFormat.OpenApi.value:
+        sFormat = ContentFormat.openapi.value if specificationPath is not None else ContentFormat.openapi_link.value
+    elif specificationFormat == ImportFormat.OpenApiJson.value:
+        sFormat = ContentFormat.openapijson.value if specificationPath is not None else ContentFormat.openapi_link.value
+    elif specificationFormat == ImportFormat.Wsdl.value:
+        sFormat = ContentFormat.wsdl.value if specificationPath is not None else ContentFormat.wsdl_link.value
+    else:
+        raise CLIError("Please provide valid value for specificationFormat: " + specificationFormat)
+
+    resource = ApiCreateOrUpdateParameter(
+        format=sFormat,
+        path=api_path
+    )
+
+    if api_revision is not None and api_id is not None:
+        api_id = api_id + ";rev=" + api_revision
+    elif api_id is None:
+        api_id = uuid.uuid4().hex
+
+    if specificationPath is not None:
+        api_file = open(specificationPath, 'r')
+        contentValue = api_file.read()
+        resource.value = contentValue
+    elif specificationUrl is not None:
+        resource.value = specificationUrl
+    else:
+        raise CLIError("Can't specify specificationUrl and specificationPath at the same time.")
+
+    if protocols is not None:
+        resource.protocols = protocols.split(',')
+
+    if service_url is not None:
```

Comment:
Based on my knowledge, this check is meaningless. The default value should be `None`. Meanwhile, I'm not familiar with APIM service, please double confirm that what you want is to create a new `ApiCreateOrUpdateParameter`.